sklearn/decomposition/dict_learning.py

```diff
@@ -282,9 +282,9 @@ def sparse_encode(X, dictionary, gram=None, cov=None, algorithm='lasso_lars',
                               check_input=False,
                               verbose=verbose)
         # This ensure that dimensionality of code is always 2,
-        # consistant with the case n_jobs > 1
+        # consistent with the case n_jobs > 1
         if code.ndim == 1:
-            code = code[np.newaxis, :]
+            code = code[:, np.newaxis]
```

Comment:
(one direction of code needs to be n_features so their could only be ambiguity when n_features=1. Both n_components=1 and n_features=1 seems odd, but I'll add some tests that make sure both work as expected and see if this has any behavior change.